lib/secure_mirror/setup.rb

```diff
@@ -28,24 +28,24 @@ def github_client
                              config: @config[:repo_types][:github])
     end
 
-    def caching_client(client)
-      conf = @config[:cache]
+    def enable_caching(client)
+      return unless client
+
+      cache = @config[:cache]
+      return client unless cache && cache[:enable]
+
       CachingMirrorClient.new(client,
-                              cache_dir: conf[:dir],
-                              default_expiration: conf[:default_expiration])
+                              cache_dir: cache[:dir],
+                              default_expiration: cache[:default_expiration])
     end
 
     def client
-      case @repo.url.downcase
-      when /github/
-        client = github_client
-      else
-        msg = "unsupported mirror type: #{@repo.url}"
```

Comment:
is this branch no longer required?

Reply:
Not here, at least. Inevitably, the mirror type is how a client is created. I've moved the responsibility of deciding on whether or not a client is necessary into the `Policy` class itself.

This is to support @paulbry's use-case where he uses the base `Policy` as sort of a "no-op" policy that will just log information about the mirror. In that case, there's no need to set up a client.

Reply:
Also, if that was to say, do we require a `case` statement at all, I'd say yes. We still need other client implementations though.